test/erlfdb_03_transaction_options_test.erl

```diff
@@ -35,5 +35,29 @@ size_limit_test() ->
     end)).
 
 
+writes_allowed_test() ->
+    Db1 = erlfdb_util:get_test_db(),
+    ?assertError(writes_not_allowed, erlfdb:transactional(Db1, fun(Tx) ->
```

Comment:
Do these work? I thought the second argument for the exception assertions had to be callable?

Reply:
Oh I see what you mean. Looked at http://erlang.org/doc/apps/eunit/chapter.html#EUnit_test_representation and the second argument can be an expression. One of the examples on the page is `?assertError(badarith, X/0)`. In the code the seem to auto-wrap it in the expression in the function automatically https://github.com/erlang/otp/blob/93a7cfeded88e89b4431b3d5609b1f22d7b63306/lib/stdlib/include/assert.hrl#L300

Reply:
Huh! Good to know!